packages/terra-form/src/Select.jsx

```diff
@@ -7,9 +7,18 @@ import './Select.scss';
 
 const propTypes = {
   /**
-   * List of choices to be selected.
+   * List of object key/value pairs for choices to be selected.
    */
-  choices: PropTypes.array.isRequired,
+  choices: PropTypes.arrayOf(PropTypes.shape({
+    value: PropTypes.oneOfType([
+      PropTypes.string,
+      PropTypes.number,
+    ]).isRequired,
+    display: PropTypes.oneOfType([
+      PropTypes.string,
+      PropTypes.number,
```

Comment:
@tbiethman We do have a case where a select box will display number options and the user will pick a number. Leaving this will save us having to convert to a string everytime, but it is not a big deal if we have to do that conversion.